### ISSUE TITLE
Completing timer utility along with histogram features.

### ISSUE THREAD

**Issue by Chryel** [gwAdvNet2015/gw-kv-store#34]
Will attempt to complete the timer utility's last timer_end function with histogram features.

**Comment by twood02** [comment]
you should integrate the timer into the simple client and you should fix the bug in the histogram related to adding values that are larger than the biggest histogram bucket (check with @akorzan if you need more details).

You should also change the client so that it can perform N requests (a command line argument).  Remember that each request should create its own socket.

Actually, since it will be difficult to test this without having an actual memcached server to send to, you could add this code to the `trivial-client` example in the repo.  If you can make the timing and histogramming work correctly for that, it should be trivial to port it over to our real client later.  This will also limit merge conflicts since the Client team is going to be making major modifications to the client's structure.  Make sense?

**Comment by Chryel** [comment]
In order to avoid the segmentation faults when adding a value that is not within the histogram's bucket range, I added a second part to the if statement in histogram_inc in order to catch any values over the range. I incremented an added value called "missed_counter" to indicate to the users that there were missed values and modified the histogram_print function accordingly.

The last function of timer_end in the timer_util file had conflicting errors with an identically named function, and was renamed to timer_end_hist. 

The "trivial" folder contains both trivial-client and trivial-server. It also contains socket_helper that is required for both files. I've tested the client by running "./trivial-client -r 1000" to simulate 1000 connections and it seems to work fine. 

Unfortunately, I've worked on this on a separate branch of adv-net-samples, so I will be making an appropriate pull request on that repo.
